### PR TITLE
Update LiteOptimizer signature after optimizer changes in TrainingTypePlugin

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow-up to #10596
Resolves a TODO in #10596

The LiteOptimizer no longer takes the accelerator as input and instead takes the Strategy. 
The accelerator is not needed, since the optimizer step is now owned by the  Strategy.

Should be merged after #10596

### Does your PR introduce any breaking changes? If yes, please list them.

None. LiteOptimizer is not a public API

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda @justusschock @awaelchli @akihironitta @carmocca

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/lite/wrappers.py [hunk 1]
```diff
@@ -30,24 +29,22 @@ def _do_nothing_closure() -> None:
...
         """
         # `__del__` is skipped in case the optimizer has implemented custom destructor logic which we would
         # not want to call on destruction of the `_LiteOptimizer
         self.__dict__ = {k: v for k, v in optimizer.__dict__.items() if k not in ("state_dict", "step", "__del__")}
         self.__class__ = type("Lite" + optimizer.__class__.__name__, (self.__class__, optimizer.__class__), {})
         self._optimizer = optimizer
-        self._accelerator = accelerator
-        # TODO (@awaelchli) refactor to take Strategy as param
-        self._strategy = self._accelerator.training_type_plugin
+        self._strategy = strategy
```
Comment (tchaton) [inline pytorch_lightning/lite/wrappers.py:47]: Neat !

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
